src/include/OpenImageIO/paramlist.h

```diff
@@ -503,4 +503,138 @@ class OIIO_UTIL_API ParamValueList : public std::vector<ParamValue> {
 
 
 
+/// A span of const ParamValue entries, that can be iterated over or searched.
+/// It's really just a cspan<ParamValue>, but with a few more handy methods.
+/// This is a convenient way to pass the contents of a ParamValueList (or its
+/// equivalent of any consecutive subarray of PV's) without any copies.
+class OIIO_UTIL_API ParamValueSpan : public cspan<ParamValue> {
+public:
+    // Note: inherits from cspan:
+    //   - size()
+    //   - data()
+    //   - operator[int]
+    //   - begin(), end(), cbegin(), cend()
+
+    ParamValueSpan(cspan<ParamValue> p)
+        : cspan<ParamValue>(p)
+    {
+    }
+
+    // Trivially make a ParamValueSpan from a ParamValueList
+    ParamValueSpan(const ParamValueList& p)
+        : cspan<ParamValue>(p)
+    {
+    }
+
+    /// Construct an span from an initializer_list.
+    constexpr ParamValueSpan(std::initializer_list<ParamValue> il)
+        : cspan<ParamValue>(il.begin(), il.size())
+    {
+    }
+
+    /// Construct from a fixed-length C array.  Template magic automatically
+    /// finds the length from the declared type of the array.
+    template<size_t N>
+    constexpr ParamValueSpan(const ParamValue (&data)[N])
+        : cspan<ParamValue>(data, N)
+    {
+    }
+
+    constexpr reference operator[](size_type idx) const
+    {
+        return cspan<ParamValue>::operator[](idx);
+    }
+
+    const_iterator find(string_view name, TypeDesc type = TypeUnknown,
+                        bool casesensitive = true) const;
+    const_iterator find(ustring name, TypeDesc type = TypeUnknown,
+                        bool casesensitive = true) const;
+
+    /// Case insensitive search for an integer, with default if not found.
+    /// Automatically will return an int even if the data is really
+    /// unsigned, short, or byte, but not float. It will retrieve from a
+    /// string, but only if the string is entirely a valid int format.
+    int get_int(string_view name, int defaultval = 0, bool casesensitive = true,
```

Comment:
Which way do you think is better in this case? I think for ImageSpec metadata, we search case-insensitively. I don't know if that's the "right" choice. But since it would be so confusing and unwise to have two separate parameters that differed only in case, I don't see an advantage to encouraging it. On the other hand, case-sensitive string compares are a lot faster. Opinions?

Reply:
I agree with your points, but I don't have a strong preference for what to default to. My main concern is that the comment and the code are currently opposite.
If I do have to pick something, I would probably default to case insensitive so it gives the results users likely expect. If anyone really needs the speed, they can still request the case sensitive fast path.